.github/workflows/haskell-ci-windows.yml

```diff
@@ -52,17 +52,21 @@ jobs:
         run: cabal freeze
 
       - uses: actions/cache@v4
-        name: Cache ~/.cabal/store
+        id: cache
+        name: ~/.cabal/store
         with:
           path: ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
-          key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+          key: ${{ matrix.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+          restore-keys: |
+            ${{ matrix.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+            ${{ matrix.os }}-${{ matrix.ghc }}-
 
       - name: Install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
         run: cabal build all --only-dependencies
```

Comment:
Is this step even necessary?

Reply:
Yes. The idea is to have cache contain built dependencies only. This way even if the project itself fails to compile, we wouldn't have to worry about re-compiling dependencies all over again.

Reply:
But won't the next step compile dependencies if the cache is not present?

Reply:
It will. But it will also compile swarm in the same step. And if compilation fails for swarm then on subsequent push, the dependencies will be compiled all over again.
Building dependencies in separate step will at least let up avoid such a case.

Reply:
@nitinprakash96 is the condition there for testing and you dont intend to merge it?